File: MusicWeb.Client/src/models/UserFavoriteArtist.js
export default class UserFavoriteArtist {
  id = null;
  userId = null;
  artistId = null;
  artist = {};
  user = {};
  artistId = null;
  constructor($data) {
    if ($data) {
      this.id = $data.id;
      this.userId = $data.userId;
      this.artistId = $data.artistId;
      this.artist = $data.artist ? $data.artist : {};
      this.user = $data.user ? $data.user : {};
    }
  }
}

File: MusicWeb.Server/MusicWeb.Models/Entities/Users/UserFavoriteArtist.cs
using MusicWeb.Models.Entities.Artists;
using MusicWeb.Models.Entities.Base;
using MusicWeb.Models.Identity;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MusicWeb.Models.Entities
{
    public class UserFavoriteArtist : BaseEntity
    {
        public DateTime FavoriteDate { get; set; }
        public string UserId { get; set; }
        public int ArtistId { get; set; }
        public virtual Artist Artist { get; set; }
        public virtual ApplicationUser User { get; set; }
    }
}
